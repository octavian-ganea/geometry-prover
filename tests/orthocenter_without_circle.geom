Description : intersection of altitudes in a triangle is a single point - the ortocenter

Hypothesis:

Let an arbitrary circle Circ(O ,radix)

Let an arbitrary point A
Let an arbitrary point B
Let an arbitrary point C

Let a 1-equation deterministic point A1 situated on projection of A on line B C
Let a 1-equation deterministic point B1 situated on projection of B on line A C
Let a 1-equation deterministic point C1 situated on projection of C on line B A

Conclusion: 

concurrent line A A1 line B B1 line C C1

/*
Let an semiarbitrary point A situated on circle Circ
Let an semiarbitrary point B situated on circle Circ
Let an semiarbitrary point C situated on circle Circ
*/
