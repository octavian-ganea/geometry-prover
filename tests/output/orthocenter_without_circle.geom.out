====== Parsing .... 
Description : intersection of altitudes in a triangle is a single point - the ortocenter 

Hypothesis:

Let an arbitrary circle Circ(O,radix)
Let an arbitrary point A
Let an arbitrary point B
Let an arbitrary point C


Let a 1-equation deterministic point A1 situated on projection of A on line B-C
Let a 1-equation deterministic point B1 situated on projection of B on line A-C
Let a 1-equation deterministic point C1 situated on projection of C on line B-A
Let a 2-equations deterministic point X13 situated on line A-A1 and line B-B1

Conclusion: points X13, C and C1 are collinear

====== Parse OK 

====== Proof of the problem:

======= Generating equations for each point ... 

========= List of arbitrary points:


========= List of semi-arbitrary points:

========= List of deterministic points:

A1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 a1  =  b * c__ + a * c__ + a__ * c - b__ * c - a__ * b - a * b__  / 2 * c__ - 2 * b__ 

 a1__  =  b__ * c + a__ * c + a * c__ - b * c__ - a * b__ - a__ * b  / 2 * c - 2 * b 


B1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 b1  =  a * c__ + b * c__ + b__ * c - a__ * c - a * b__ - a__ * b  / 2 * c__ - 2 * a__ 

 b1__  =  a__ * c + b__ * c + b * c__ - a * c__ - a__ * b - a * b__  / 2 * c - 2 * a 


C1 is a deterministic point defined by equations (After substitutions and simplifications):
 
 c1  =  a * b__ + b__ * c + b * c__ - a__ * b - a__ * c - a * c__  / 2 * b__ - 2 * a__ 

 c1__  =  a__ * b + b * c__ + b__ * c - a * b__ - a * c__ - a__ * c  / 2 * b - 2 * a 


X13 is defined by equation: a__ * x13 + a * a1__ + a1 * x13__ - a1__ * x13 - a * x13__ - a1 * a__ = 0
         and by equation: b__ * x13 + b * b1__ + b1 * x13__ - b1__ * x13 - b * x13__ - b1 * b__ = 0
After substitutions and simplifications:
 x13  =  7 * a * a__ * b * b__^2 * c^2 + 2 * a * b * b__^2 * c^2 * c__ + 5 * a^3 * b__^2 * c * c__ + a^2 * a__ * b__^2 * c^2 + 2 * a * b__^3 * c^3 + a^2 * b^2 * b__ * c__^2 + 7 * a^2 * b * b__ * c * c__^2 + 4 * a^3 * b * b__^2 * c__ + a^3 * a__ * b__^2 * c + a^4 * b__^3 + 2 * a^3 * a__ * b * b__ * c__ + 3 * a^4 * b__ * c__^2 + a^2 * b__ * c^2 * c__^2 + 3 * a * b^2 * c * c__^3 + 2 * a^3 * b * c__^3 + a^3 * c * c__^3 + a^3 * a__ * c * c__^2 + 5 * a * a__ * b^3 * c__^2 + 4 * a__ * b^3 * c * c__^2 + 7 * a * a__^2 * b^2 * c * c__ + a__^2 * b^2 * c^2 * c__ + 2 * a^2 * a__ * b * c * c__^2 + a__^2 * b * c^3 * c__ + 2 * a * a__^2 * b^2 * b__ * c + 5 * a__^2 * b * b__ * c^3 + 4 * a * a__^2 * b__ * c^3 + 3 * a__ * b__^2 * c^4 + a * a__^3 * b^3 + 2 * a__^3 * b^3 * c + a^2 * a__^2 * b^2 * c__ + 3 * a * a__^3 * b * c^2 + 2 * a^2 * a__^2 * b * b__ * c + a^2 * a__^2 * c^2 * c__ + a__^3 * c^4 + a * a__^2 * b^3 * b__ + a * b^3 * b__ * c__^2 + 3 * a^2 * b * b__^3 * c + 2 * a * a__ * b * c^2 * c__^2 + 2 * a * a__ * b__ * c^3 * c__ + 3 * a__^2 * b^4 * c__ + b^4 * c__^3 + 2 * a * b^2 * b__^2 * c * c__ + b * b__^2 * c^3 * c__ + 2 * a__ * b^3 * b__ * c * c__ + b * b__^3 * c^3 + b^2 * b__ * c^2 * c__^2 + a^2 * a__ * b^2 * b__^2 + a__ * b^2 * b__^2 * c^2 - 2 * a^2 * a__ * b * b__^2 * c - 7 * a^2 * b * b__^2 * c * c__ - a^2 * b__^2 * c^2 * c__ - 5 * a * a__ * b__^2 * c^3 - 2 * a^3 * b__^3 * c - 2 * a * b^2 * b__ * c * c__^2 - 5 * a^3 * b * b__ * c__^2 - a^3 * a__ * b * b__^2 - 3 * a^4 * b__^2 * c__ - 4 * a^3 * b__ * c * c__^2 - 2 * a^3 * a__ * b__ * c * c__ - 2 * a * b * b__ * c^2 * c__^2 - a * b__^2 * c^3 * c__ - 3 * a^2 * b * c * c__^3 - a^3 * a__ * b * c__^2 - a^4 * c__^3 - a^2 * a__ * c^2 * c__^2 - 4 * a * a__^2 * b^3 * c__ - 5 * a__^2 * b^3 * c * c__ - 7 * a * a__ * b^2 * c * c__^2 - a__ * b^2 * c^2 * c__^2 - 2 * a__ * b * b__ * c^3 * c__ - a__^2 * b^2 * b__ * c^2 - 7 * a * a__^2 * b * b__ * c^2 - 4 * a__ * b * b__^2 * c^3 - 3 * a__^2 * b__ * c^4 - 3 * a * a__^3 * b^2 * c - a^2 * a__^2 * b^2 * b__ - 2 * a^2 * a__^2 * b * c * c__ - 2 * a__^3 * b * c^3 - a * a__^2 * c^3 * c__ - a * a__^3 * c^3 - a^2 * a__^2 * b__ * c^2 - 2 * a * a__ * b^3 * b__ * c__ - a^3 * b * b__^3 - 3 * a * b * b__^3 * c^2 - a^2 * a__ * b^2 * c__^2 - 2 * a * b^3 * c__^3 - 2 * a * a__^2 * b * c^2 * c__ - 3 * a__ * b^4 * c__^2 - a^2 * b^2 * b__^2 * c__ - b^2 * b__^2 * c^2 * c__ - b^3 * c * c__^3 - a__^2 * b^3 * b__ * c - b^3 * b__ * c * c__^2 - b__^3 * c^4 - a__^3 * b^4 - 2 * a * a__ * b^2 * b__^2 * c  / 3 * a * a__^2 * b^2 * b__ + 6 * a__ * b^2 * b__ * c * c__ + 6 * a^2 * a__ * b * b__ * c__ + 6 * a__^2 * b * b__ * c^2 + 6 * a * a__ * b * b__^2 * c + 6 * a * a__ * b__ * c^2 * c__ + 3 * a * a__^2 * b__ * c^2 + 3 * a^2 * a__ * b__^2 * c + 3 * a__ * b__^2 * c^3 + 6 * a * a__ * b^2 * c__^2 + 3 * a__ * b^2 * c * c__^2 + 6 * a * a__^2 * b * c * c__ + 3 * a__^2 * b * c^2 * c__ + 3 * a^2 * a__ * c * c__^2 + 3 * b * b__^2 * c^2 * c__ + 6 * a^2 * b__^2 * c * c__ + 3 * a * b__^3 * c^2 + 3 * a * b^2 * b__ * c__^2 + 6 * a * b * b__ * c * c__^2 + 3 * a^2 * b * b__^2 * c__ + a^3 * b__^3 + 3 * a^3 * b__ * c__^2 + 3 * a^2 * b * c__^3 + 3 * a__^2 * b^3 * c__ + b^3 * c__^3 + 3 * a__^3 * b^2 * c + a__^3 * c^3 - 3 * a__^2 * b^2 * b__ * c - 6 * a * a__ * b^2 * b__ * c__ - 6 * a * a__^2 * b * b__ * c - 3 * a^2 * a__ * b * b__^2 - 3 * a__ * b * b__^2 * c^2 - 6 * a^2 * a__ * b__ * c * c__ - 3 * a__^2 * b__ * c^3 - 6 * a * a__ * b__^2 * c^2 - 3 * a * a__^2 * b^2 * c__ - 6 * a__^2 * b^2 * c * c__ - 3 * a^2 * a__ * b * c__^2 - 3 * a * a__^2 * c^2 * c__ - 6 * a * b * b__^2 * c * c__ - 3 * a * b__^2 * c^2 * c__ - 3 * a^2 * b__^3 * c - b__^3 * c^3 - 3 * b^2 * b__ * c * c__^2 - 6 * a^2 * b * b__ * c__^2 - 3 * a^3 * b__^2 * c__ - 3 * a^2 * b__ * c * c__^2 - 3 * a * b^2 * c__^3 - a^3 * c__^3 - 6 * a * a__ * b * c * c__^2 - 3 * a__ * b^3 * c__^2 - 6 * a__ * b * b__ * c^2 * c__ - a__^3 * b^3 - 3 * a__^3 * b * c^2
 x13__  =  7 * a * a__ * b^2 * b__ * c__^2 + 2 * a__ * b^2 * b__ * c * c__^2 + 5 * a__^3 * b^2 * c * c__ + a * a__^2 * b^2 * c__^2 + 2 * a__ * b^3 * c__^3 + a__^2 * b * b__^2 * c^2 + 7 * a__^2 * b * b__ * c^2 * c__ + 4 * a__^3 * b^2 * b__ * c + a * a__^3 * b^2 * c__ + a__^4 * b^3 + 2 * a * a__^3 * b * b__ * c + 3 * a__^4 * b * c^2 + a__^2 * b * c^2 * c__^2 + 3 * a__ * b__^2 * c^3 * c__ + 2 * a__^3 * b__ * c^3 + a__^3 * c^3 * c__ + a * a__^3 * c^2 * c__ + 5 * a * a__ * b__^3 * c^2 + 4 * a * b__^3 * c^2 * c__ + 7 * a^2 * a__ * b__^2 * c * c__ + a^2 * b__^2 * c * c__^2 + 2 * a * a__^2 * b__ * c^2 * c__ + a^2 * b__ * c * c__^3 + 2 * a^2 * a__ * b * b__^2 * c__ + 5 * a^2 * b * b__ * c__^3 + 4 * a^2 * a__ * b * c__^3 + 3 * a * b^2 * c__^4 + a^3 * a__ * b__^3 + 2 * a^3 * b__^3 * c__ + a^2 * a__^2 * b__^2 * c + 3 * a^3 * a__ * b__ * c__^2 + 2 * a^2 * a__^2 * b * b__ * c__ + a^2 * a__^2 * c * c__^2 + a^3 * c__^4 + a^2 * a__ * b * b__^3 + a__ * b * b__^3 * c^2 + 3 * a__^2 * b^3 * b__ * c__ + 2 * a * a__ * b__ * c^2 * c__^2 + 2 * a * a__ * b * c * c__^3 + 3 * a^2 * b__^4 * c + b__^4 * c^3 + 2 * a__ * b^2 * b__^2 * c * c__ + b^2 * b__ * c * c__^3 + 2 * a * b * b__^3 * c * c__ + b^3 * b__ * c__^3 + b * b__^2 * c^2 * c__^2 + a * a__^2 * b^2 * b__^2 + a * b^2 * b__^2 * c__^2 - 2 * a * a__^2 * b^2 * b__ * c__ - 7 * a__^2 * b^2 * b__ * c * c__ - a__^2 * b^2 * c * c__^2 - 5 * a * a__ * b^2 * c__^3 - 2 * a__^3 * b^3 * c__ - 2 * a__ * b * b__^2 * c^2 * c__ - 5 * a__^3 * b * b__ * c^2 - a * a__^3 * b^2 * b__ - 3 * a__^4 * b^2 * c - 4 * a__^3 * b * c^2 * c__ - 2 * a * a__^3 * b * c * c__ - 2 * a__ * b * b__ * c^2 * c__^2 - a__ * b^2 * c * c__^3 - 3 * a__^2 * b__ * c^3 * c__ - a * a__^3 * b__ * c^2 - a__^4 * c^3 - a * a__^2 * c^2 * c__^2 - 4 * a^2 * a__ * b__^3 * c - 5 * a^2 * b__^3 * c * c__ - 7 * a * a__ * b__^2 * c^2 * c__ - a * b__^2 * c^2 * c__^2 - 2 * a * b * b__ * c * c__^3 - a^2 * b * b__^2 * c__^2 - 7 * a^2 * a__ * b * b__ * c__^2 - 4 * a * b^2 * b__ * c__^3 - 3 * a^2 * b * c__^4 - 3 * a^3 * a__ * b__^2 * c__ - a^2 * a__^2 * b * b__^2 - 2 * a^2 * a__^2 * b__ * c * c__ - 2 * a^3 * b__ * c__^3 - a^2 * a__ * c * c__^3 - a^3 * a__ * c__^3 - a^2 * a__^2 * b * c__^2 - 2 * a * a__ * b * b__^3 * c - a__^3 * b^3 * b__ - 3 * a__ * b^3 * b__ * c__^2 - a * a__^2 * b__^2 * c^2 - 2 * a__ * b__^3 * c^3 - 2 * a^2 * a__ * b__ * c * c__^2 - 3 * a * b__^4 * c^2 - a__^2 * b^2 * b__^2 * c - b^2 * b__^2 * c * c__^2 - b__^3 * c^3 * c__ - a^2 * b * b__^3 * c__ - b * b__^3 * c^2 * c__ - b^3 * c__^4 - a^3 * b__^4 - 2 * a * a__ * b^2 * b__^2 * c__  / 3 * a^2 * a__ * b * b__^2 + 6 * a * b * b__^2 * c * c__ + 6 * a * a__^2 * b * b__ * c + 6 * a^2 * b * b__ * c__^2 + 6 * a * a__ * b^2 * b__ * c__ + 6 * a * a__ * b * c * c__^2 + 3 * a^2 * a__ * b * c__^2 + 3 * a * a__^2 * b^2 * c__ + 3 * a * b^2 * c__^3 + 6 * a * a__ * b__^2 * c^2 + 3 * a * b__^2 * c^2 * c__ + 6 * a^2 * a__ * b__ * c * c__ + 3 * a^2 * b__ * c * c__^2 + 3 * a * a__^2 * c^2 * c__ + 3 * b^2 * b__ * c * c__^2 + 6 * a__^2 * b^2 * c * c__ + 3 * a__ * b^3 * c__^2 + 3 * a__ * b * b__^2 * c^2 + 6 * a__ * b * b__ * c^2 * c__ + 3 * a__^2 * b^2 * b__ * c + a__^3 * b^3 + 3 * a__^3 * b * c^2 + 3 * a__^2 * b__ * c^3 + 3 * a^2 * b__^3 * c + b__^3 * c^3 + 3 * a^3 * b__^2 * c__ + a^3 * c__^3 - 3 * a^2 * b * b__^2 * c__ - 6 * a * a__ * b * b__^2 * c - 6 * a^2 * a__ * b * b__ * c__ - 3 * a * a__^2 * b^2 * b__ - 3 * a * b^2 * b__ * c__^2 - 6 * a * a__^2 * b * c * c__ - 3 * a^2 * b * c__^3 - 6 * a * a__ * b^2 * c__^2 - 3 * a^2 * a__ * b__^2 * c - 6 * a^2 * b__^2 * c * c__ - 3 * a * a__^2 * b__ * c^2 - 3 * a^2 * a__ * c * c__^2 - 6 * a__ * b^2 * b__ * c * c__ - 3 * a__ * b^2 * c * c__^2 - 3 * a__^2 * b^3 * c__ - b^3 * c__^3 - 3 * b * b__^2 * c^2 * c__ - 6 * a__^2 * b * b__ * c^2 - 3 * a__^3 * b^2 * c - 3 * a__^2 * b * c^2 * c__ - 3 * a__ * b__^2 * c^3 - a__^3 * c^3 - 6 * a * a__ * b__ * c^2 * c__ - 3 * a * b__^3 * c^2 - 6 * a * b * b__ * c * c__^2 - a^3 * b__^3 - 3 * a^3 * b__ * c__^2 

=========== Equations were generated successfully ========

======= The conclusion may be correct only unde the following restrictions:
 -- points B and A must be different
 -- points C and A must be different
 -- points C and B must be different

====== Trying to solve the goal of the problem ...
Conclusion is equivalent to equation: 
c__ * x13 + c * c1__ + c1 * x13__ - c1__ * x13 - c * x13__ - c1 * c__ = 0

Attempt to prove it by substitutions and simplifications ....
 
  !!!!!!!!!!!!!!!!!!!!!
  !!!!!! SUCCESS !!!!!!
  !!!!!!!!!!!!!!!!!!!!!

===== Problem is correct. Proof ends here. QED ====

